src/script/lua_api/l_object.cpp

```diff
@@ -1246,12 +1246,42 @@ int ObjectRef::l_get_look_dir(lua_State *L)
 	if (playersao == nullptr)
 		return 0;
 
-	float pitch = playersao->getRadLookPitchDep();
-	float yaw = playersao->getRadYawDep();
-	v3f v(std::cos(pitch) * std::cos(yaw), std::sin(pitch), std::cos(pitch) *
-		std::sin(yaw));
+	push_v3f(L, playersao->getLookDir());
+	return 1;
+}
+
+// get_point_screen_pos(self)
+int ObjectRef::l_get_point_screen_pos(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == nullptr)
+		return 0;
+
+	push_v2f(L, player->pointer_pos);
+	return 1;
+}
+
+// get_point_dir(self)
+int ObjectRef::l_get_point_dir(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	PlayerSAO* playersao = getplayersao(ref);
+	if (playersao == nullptr)
+		return 0;
+
+	RemotePlayer *player = playersao->getPlayer();
+
+	if (player->pointer_pos.X == 0.5f && player->pointer_pos.Y == 0.5f) {
```

Comment:
This `if` block must trigger if a crosshair is used, since `point_pitch` and `point_yaw` will contain garbage values (zeroes) in this case. `pointer_pos.X` and `pointer_pos.Y` are set to literal `0.5f` if a crosshair is used, so comparing them with `==` should be fine.

However, I'm not sure if it's guaranteed to be fine since these values are serialized and sent over the network. I could add a small tolerance just to be sure.